src/products/components/ProductStocks/ProductStocks.tsx

```diff
@@ -45,8 +45,55 @@ export interface ProductStocksProps {
   onWarehouseConfigure: () => void;
   fetchMoreWarehouses: () => void;
   hasMoreWarehouses: boolean;
+  isCreate: boolean;
 }
 
+const WarehouseInformationMessage = ({
+  isCreate,
+  hasVariants,
+  hasWarehouses,
+  onWarehouseConfigure,
+}: {
+  isCreate: boolean;
+  hasVariants: boolean;
+  hasWarehouses: boolean;
+  onWarehouseConfigure: () => void;
+}) => {
+  if (isCreate) {
+    const message = hasVariants
+      ? messages.warehouseMessageVariantOnCreate
+      : messages.warehouseMessageProductOnCreate;
+
+    return (
+      <Text color="default2">
+        <FormattedMessage {...message} />
+      </Text>
+    );
+  }
+
+  return (
+    !hasWarehouses && (
+      <Text color="default2">
+        {hasVariants ? (
+          <FormattedMessage
+            {...messages.configureWarehouseForVariant}
+            values={{
+              a: chunks => <Link onClick={onWarehouseConfigure}>{chunks}</Link>,
+            }}
+          />
+        ) : (
+          <FormattedMessage
+            {...messages.configureWarehouseForProduct}
+            values={{
+              a: chunks => <Link onClick={onWarehouseConfigure}>{chunks}</Link>,
+            }}
+          />
+        )}
+      </Text>
+    )
+  );
+};
```

Comment:
I'd recommend to write unite test for this
```suggestion
const WarehouseInformationMessage = ({
  isCreate,
  hasVariants,
  hasWarehouses,
  onWarehouseConfigure,
}: {
  isCreate: boolean;
  hasVariants: boolean;
  hasWarehouses: boolean;
  onWarehouseConfigure: () => void;
}) => {
  if (isCreate) {
    const message = hasVariants
      ? messages.warehouseMessageVariantOnCreate
      : messages.warehouseMessageProductOnCreate;

    return (
      <Text color="default2">
        <FormattedMessage {...message} />
      </Text>
    );
  }

  if (hasWarehouses) {
    return null;
  }

  return (
    <Text color="default2">
      {hasVariants ? (
        <FormattedMessage
          {...messages.configureWarehouseForVariant}
          values={{
            a: chunks => <Link onClick={onWarehouseConfigure}>{chunks}</Link>,
          }}
        />
      ) : (
        <FormattedMessage
          {...messages.configureWarehouseForProduct}
          values={{
            a: chunks => <Link onClick={onWarehouseConfigure}>{chunks}</Link>,
          }}
        />
      )}
    </Text>
  );
};
```